### PR TITLE
The multipackage: Breaking apart the NPM export

### PR DESCRIPTION
**Pull request by ericvicenti**
[Read the RFC here](https://github.com/react-navigation/rfcs/blob/925ab7e43ae419393c8dbc3ed0bd93261de897ed/text/0007-multi-package.md)

### CODE REVIEW

**Comment by dantman** [comment]
I know it will slightly discourage user packages. But if you're going the multiple-package approach I'd recommend going with `@react-navigation/core`, etc. That way it's clear which packages are maintained by the core community and which are maintained by 3rd parties. And a poorly written and abandoned 3rd party `react-navigation-dialog-navigator` doesn't block the core react-navigation community from creating a good DialogNavigator.

**Comment by ericvicenti** [comment]
@dantman, I think thats a good idea!

**Comment by ericvicenti** [comment]
So, I have been making progress on this in the past few weeks, even though we seem to be in a maintainer deadlock on this issue. People generally want this, but aren't willing to accept the increased complexity or workflow overhead. So I've been forced to find an incremental approach until we find a  path that makes everybody happy.

tldr:, I've created a way to import the react-navigation source code into a different repository. This allows me to test and experiment with all of the different packages in the same environment (including web+server rendering).

https://github.com/ericvicenti/universe

Soon I'll get the playground and tests copied over there as well. Each package lives in its own folder, which I plan to publish independently on github via a sub-repository.

So that means that the main `react-navigation` repo can continue to exist, while some maintainers such as myself will start developing in the new environment. No code will go stale, because there is an automatic way to pull the latest code out of the main legacy repo. It will still be the source of truth for core modules.

We will soon be able to publish these smaller packages to NPM, so that people can simply use the parts of react-nav that they need.

Let me know if anybody wants to help with this. I probably won't be landing this RFC because I couldn't get enough agreement for either of the two proposed directions. So I'll make some independent progress on this, and leave the main repository as-is, until we get the other workflow figured out.

**Comment by brentvatne** [comment]
this is done now! we skipped the RFC process here a bit though. we settled on @react-navigation/core, @react-navigation/web and @react-navigation/native for the platform-agnostic code, web container, and native container respectively. navigators will continue to live in their own repos at react-navigation-stack, react-navigation-tabs, and react-navigation-drawer, so they are peers with other navigators built by the community. react-navigation will continue to pull in the correct packages to work seamlessly in react-native on ios and android in order to minimize disruption - this may be reconsidered in the future.

with this comment i'll close this PR because it's not totally accurate to what @ericvicenti and i settled on.
